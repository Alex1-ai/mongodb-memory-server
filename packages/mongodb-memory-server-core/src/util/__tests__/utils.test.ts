import { Stats, promises } from 'fs';
import * as utils from '../utils';

describe('utils', () => {
  describe('uriTemplate', () => {
    it('should not add "?" without query options', () => {
      expect(utils.uriTemplate('0.0.0.0', 1001, 'somedb')).toEqual('mongodb://0.0.0.0:1001/somedb');
    });

    it('should add "?" with query options', () => {
      expect(utils.uriTemplate('0.0.0.0', 1001, 'somedb', ['option1=1', 'option2=2'])).toEqual(
        'mongodb://0.0.0.0:1001/somedb?option1=1&option2=2'
      );
    });
  });

  describe('pathExists', () => {
    it('should return true if there are stats', async () => {
      jest.spyOn(promises, 'stat').mockResolvedValueOnce(new Stats());
      await expect(utils.pathExists('/some/path')).resolves.toEqual(true);
    });

    it('should return false if there are no stats', async () => {
      const retError = new Error();
      // @ts-expect-error because there is no FSError, or an error with an "code" property - but still being used
      retError.code = 'ENOENT';
      jest.spyOn(promises, 'stat').mockRejectedValueOnce(retError);
      await expect(utils.pathExists('/some/path')).resolves.toEqual(false);
    });
  });

  describe('statPath', () => {
    it('should throw an error if code is not "ENOENT"', async () => {
      const retError = new Error();
      // @ts-expect-error because there is no FSError, or an error with an "code" property - but still being used
      retError.code = 'EPERM';
      jest.spyOn(promises, 'stat').mockRejectedValueOnce(retError);
      await expect(utils.statPath('/some/path')).rejects.toThrowError(retError);
    });
  });

  describe('assertion', () => {
    it('should throw default error if none provided', () => {
      expect.assertions(1);
      try {
        utils.assertion(false);
        fail('Expected Assertion to Throw');
      } catch (err) {
        expect(err.message).toEqual('Assert failed - no custom error [E019]');
      }
    });
  });
});
