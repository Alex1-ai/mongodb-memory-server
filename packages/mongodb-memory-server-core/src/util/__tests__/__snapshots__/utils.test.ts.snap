// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils assertion should throw default error if none provided 1`] = `"Assert failed - no custom error"`;
